src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -625,6 +642,16 @@ public static unsafe string Join<T>(string? separator, IEnumerable<T> values)
 
         public static string Join(string? separator, IEnumerable<string?> values)
```

Comment:
Did we consider also doing these checks in the:
```C#
public static string Join(char separator, IEnumerable<T> values)
```
overload?  I'm a little concerned now without that we've created a cliff.  Imagine someone writes:
```C#
string result = string.Join(",", strings);
```
and then realizes they can reduce that to:
```C#
string result = string.Join(',', strings);
```
They just lost this optimization.